verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -217,17 +212,27 @@ void IndexingFactsTreeExtractor::ExtractModuleHeader(
     return;
   }
 
-  Visit(*port_list);
+  bool has_propagated_type = false;
+  for (const auto& port : port_list->children()) {
+    if (port.get()->Kind() == verible::SymbolKind::kLeaf) continue;
+
+    const SyntaxTreeNode& port_node = verible::SymbolCastToNode(*port.get());
+    const auto tag = static_cast<verilog::NodeEnum>(port_node.Tag().tag);
+
+    if (tag == NodeEnum::kPortDeclaration) {
+      has_propagated_type = true;
+      ExtractModulePort(port_node, has_propagated_type);
+    } else if (tag == NodeEnum::kPort) {
+      ExtractModulePort(GetPortReferenceFromPort(port_node),
+                        has_propagated_type);
+    }
+  }
 }
 
 void IndexingFactsTreeExtractor::ExtractModulePort(
-    const SyntaxTreeNode& module_port_node) {
+    const SyntaxTreeNode& module_port_node, bool has_propagated_type = 0) {
```

Comment:
If you always invoke this method with the parameter, there's no need to provide a default value.